packages/js/product-editor/src/components/pricing-block/edit.tsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { createElement, useContext, Fragment } from '@wordpress/element';
+import interpolateComponents from '@automattic/interpolate-components';
+import { Link } from '@woocommerce/components';
+import { useBlockProps } from '@wordpress/block-editor';
+import { useEntityProp } from '@wordpress/core-data';
+import { BlockAttributes } from '@wordpress/blocks';
+import { CurrencyContext } from '@woocommerce/currency';
+import { getSetting } from '@woocommerce/settings';
+import { recordEvent } from '@woocommerce/tracks';
+import {
+	BaseControl,
+	// @ts-expect-error `__experimentalInputControl` does exist.
+	__experimentalInputControl as InputControl,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { formatCurrencyDisplayValue } from '../../utils';
+import { useCurrencyInputProps } from '../../hooks/use-currency-input-props';
+
+export function Edit( { attributes }: { attributes: BlockAttributes } ) {
+	const blockProps = useBlockProps();
+	const { name, label, showPricingSection = false } = attributes;
+	const [ regularPrice, setRegularPrice ] = useEntityProp< string >(
+		'postType',
+		'product',
+		name
+	);
+	const context = useContext( CurrencyContext );
+	const { getCurrencyConfig, formatAmount } = context;
+	const currencyConfig = getCurrencyConfig();
+	const inputProps = useCurrencyInputProps( {
```

Comment:
What do you suggest if we need to extends (not overrides) the `onFocus` or `onKeyUp` events coming from `useCurrencyInputProps`?

Reply:
Provided it as optional options: https://github.com/woocommerce/woocommerce/pull/37211/commits/5481c9a85233e5a217ced4947b24669649e7a640